pkg/artifacts/artifacts_storage.go

```diff
@@ -53,11 +53,54 @@ type ArtifactStorageInterface interface {
 	StorageBasePath(pr *v1alpha1.PipelineRun) string
 }
 
+// ArtifactStorageNone is used when no storage is needed.
+type ArtifactStorageNone struct{}
+
+// GetCopyToStorageFromContainerSpec returns no containers because none are needed.
+func (a *ArtifactStorageNone) GetCopyToStorageFromContainerSpec(name, sourcePath, destinationPath string) []corev1.Container {
+	return nil
+}
+
+// GetCopyFromStorageToContainerSpec returns no containers because none are needed.
+func (a *ArtifactStorageNone) GetCopyFromStorageToContainerSpec(name, sourcePath, destinationPath string) []corev1.Container {
+	return nil
+}
+
+// GetSecretsVolumes returns no volumes because none are needed.
+func (a *ArtifactStorageNone) GetSecretsVolumes() []corev1.Volume {
+	return nil
+}
+
+// GetType returns the string "none" to indicate this is the None storage type.
+func (a *ArtifactStorageNone) GetType() string {
+	return "none"
+}
+
+// StorageBasePath returns an empty string because no storage is being used and so
+// there is no path that resources should be copied from / to.
+func (a *ArtifactStorageNone) StorageBasePath(pr *v1alpha1.PipelineRun) string {
+	return ""
+}
+
 // InitializeArtifactStorage will check if there is there is a
-// bucket configured or create a PVC
-func InitializeArtifactStorage(pr *v1alpha1.PipelineRun, c kubernetes.Interface, logger *zap.SugaredLogger) (ArtifactStorageInterface, error) {
+// bucket configured, create a PVC or return nil if no storage is required.
+func InitializeArtifactStorage(pr *v1alpha1.PipelineRun, p *v1alpha1.Pipeline, c kubernetes.Interface, logger *zap.SugaredLogger) (ArtifactStorageInterface, error) {
```

Comment:
nit: Should we just pass the list of Tasks here ? (as it's the only thing we really need from `Pipeline`)

Reply:
Great idea!